core/mutex.c

```diff
@@ -50,6 +50,11 @@
 static inline __attribute__((always_inline)) void _block(mutex_t *mutex,
                                                          unsigned irq_state)
 {
+#if IS_USED(MODULE_CORE_MUTEX_DEBUG)
+    printf("[mutex] waiting for thread %" PRIkernel_pid " (pc = 0x%" PRIxTXTPTR
```

Comment:
This is intentionally not a `DEBUG(...)`, to print this in any case if we don´t want to be confronted with too many other debug messages, right?

Reply:
Yes, the idea is that `ENABLE_DEBUG` will be set to `1` to debug `core/mutex`, while the module `core_mutex_debug` aids debugging application code (where one does not suspect a bug in the mutex implementation).